### ISSUE TITLE
Make credentials resolution API private

### ISSUE THREAD

**Issue by ElenaKhaustova** [kedro-org/kedro#4259]
## Description
Currently, we pass credentials to the catalog separately from the rest configuration: https://github.com/kedro-org/kedro/blob/a5d9bb40380c598bf7d03cb16623026892844ed4/kedro/framework/context/context.py#L236

Then we resolve credentials at the `CatalogConfigResolver` level: https://github.com/kedro-org/kedro/blob/a5d9bb40380c598bf7d03cb16623026892844ed4/kedro/io/catalog_config_resolver.py#L33

We suggest to move this resolution to the config component, then:

1. We simplify `CatalogConfigResolver`, so it only resolves patterns and there's no confusion when using "resolve" term;
2. All the configuration resolution will be handled in one place by `config` module

## Context
This should be done as a preceding step to https://github.com/kedro-org/kedro/issues/3811

**Comment by datajoely** [comment]
Such a welcome move

**Comment by datajoely** [comment]
Little hack to make my MVP resolver mask credentials in the `__repr__`
<img width="861" alt="image" src="https://github.com/user-attachments/assets/bbc777d0-8c14-4726-91ae-578b237f1cb7">

**Comment by astrojuanlu** [comment]
@ElenaKhaustova do you know if this needs a breaking change?

**Comment by ElenaKhaustova** [comment]
> [@ElenaKhaustova](https://github.com/ElenaKhaustova) do you know if this needs a breaking change?

Yes, because currently we have a public API at the level of `CatalogConfigResolver ` for that

**Comment by ElenaKhaustova** [comment]
@merelcht, @astrojuanlu 

Since we do not yet have a concrete solution for https://github.com/kedro-org/kedro/discussions/4320, we agreed to make credential resolution API private at the catalog level to give ourselves room for tackling the original issue properly after 1.0.0.

**Comment by ElenaKhaustova** [comment]
Solved in [#4717](https://github.com/kedro-org/kedro/pull/4717)
